src/main/java/dk/cngroup/lentils/service/GameLogicService.java

```diff
@@ -1,24 +1,53 @@
 package dk.cngroup.lentils.service;
 
+import dk.cngroup.lentils.entity.CypherStatus;
 import dk.cngroup.lentils.entity.FinalPlace;
+import dk.cngroup.lentils.entity.Status;
+import dk.cngroup.lentils.entity.Team;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
 import java.time.LocalDateTime;
+import java.util.List;
 
-@Component
+@Service
 public class GameLogicService {
 
     private final FinalPlaceService finalPlaceService;
+    private final StatusService statusService;
 
     @Autowired
-    public GameLogicService(final FinalPlaceService finalPlaceService) {
+    public GameLogicService(final FinalPlaceService finalPlaceService,
+                            final StatusService statusService) {
         this.finalPlaceService = finalPlaceService;
+        this.statusService = statusService;
     }
 
     public boolean isGameInProgress() {
         final FinalPlace finalPlace = finalPlaceService.getFinalPlace();
 
         return finalPlace.getOpeningTime() != null && finalPlace.getOpeningTime().isAfter(LocalDateTime.now());
     }
+
+    public boolean allowPlayersToViewFinalPlace(final Team team) {
+        if (passedTimeToViewFinalPlace() || passedAllCyphers(team)) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean passedAllCyphers(final Team team) {
+        List<Status> statusesOfTeam = statusService.getAllByTeam(team);
+        Long numberOfStatusPendingByTeam = statusesOfTeam.stream()
+                .filter(status -> status.getCypherStatus().equals(CypherStatus.PENDING)).count();
+        if (numberOfStatusPendingByTeam > 0) {
```

Comment:
To same tady. Otocis logiku a das return:

return numberOfStatusPendingByTeam <= 0;